man1/aur-build.1

```diff
@@ -172,6 +172,20 @@ subdirectory that serves as template for user containers (named after
 or
 .B /copy
 if unset).
+.IP
```

Comment:
I wonder if this is not too much of an edge case to document. I was more thinking of documenting that users may want a dedicated directory (such as the mentioned `/tmp/something`) for the container, if that's not obvious already.

Reply:
Well, beside `/` every file-system is, or at leas should be, mounted with `nosuid`.
It's a default option for every `tmpfs` on Arch, which basically prevent user from building in ram with `-D /tmp` without preforming the remount first, or creating temporary `tmpfs` with `suid` to facilitate `chroot`.

Please excuse my boldness, but would you consider adding `--ram` switch to `aur-build` to automatically create temporary ram drive with `suid` to host chroot build?
```diff
--- a/lib/aur-build
+++ b/lib/aur-build
@@ -9,7 +9,7 @@ XDG_CONFIG_HOME=${XDG_CONFIG_HOME:-$HOME/.config}
 PS4='+(${BASH_SOURCE}:${LINENO}): ${FUNCNAME[0]:+${FUNCNAME[0]}(): }'
 
 # default options
-chroot=0 no_sync=0 overwrite=0 sign_pkg=0 run_pkgver=0 results=0 prefix=aur
+chroot=0 no_sync=0 overwrite=0 sign_pkg=0 run_pkgver=0 ram_drive=0 results=0 prefix=aur
 
 # default arguments
 chroot_args=()
@@ -36,6 +36,9 @@ trap_exit() {
         printf >&2 'AUR_DEBUG: %s: temporary files at %s\n' "$argv0" "$tmp"
         printf >&2 'AUR_DEBUG: %s: temporary files at %s\n' "$argv0" "$var_tmp"
     fi
+    if (( ram_drive )); then
+        sudo umount $(ram_drive_path)
+    fi
 }
 
 usage() {
@@ -97,6 +100,9 @@ while true; do
         --pkgver)
             run_pkgver=1
             makepkg_args+=(--holdver) ;;
+        --ram)
+            ram_drive=1
+            chroot=1 ;;
         --results)
             shift; results_file=$1 ;;
         --root)
@@ -200,6 +206,13 @@ if [[ -v results_file ]]; then
     results=1
 fi
 
+if (( ram_drive )); then
+    # Create temporary ram drive
+    ram_drive_path=$(mktemp -d --tmpdir "aurutils-aurbuild.XXXXXXXX") || exit
+    sudo mount -t tmpfs -o suid tmpfs "$ram_drive_path"
+    chroot_args+=(-D "$ram_drive_path")
+fi
+
 if (( chroot )); then
     # Defaults to /usr/share/devtools/pacman-aur.conf
     chroot_args+=(--prefix "$prefix")

Reply:
`nosuid` doesn't prevent building in a tmpfs (`/tmp`).  Nor does it prevent the creation of `suid` binaries:

```
% /tmp touch foo

% /tmp chmod -c 4755 foo
mode of 'foo' changed from 0644 (rw-r--r--) to 4755 (rwsr-xr-x)

% /tmp stat foo
  File: foo
  Size: 0               Blocks: 0          IO Block: 4096   regular empty file
Device: 35h/53d Inode: 143782      Links: 1
Access: (4755/-rwsr-xr-x)  Uid: ( 1000/ earnest)   Gid: (  100/   users)
Access: 2020-02-03 16:31:20.745049394 +0000
Modify: 2020-02-03 16:31:20.745049394 +0000
Change: 2020-02-03 16:31:27.305066277 +0000
 Birth: -
```

Reply:
>  would you consider adding --ram switch

As I already said: No. This is far out of scope for anything aur-chroot does. If you want that functionality, it's easy enough to add it to a wrapper script that calls aur-chroot. 

> nosuid doesn't prevent building in a tmpfs (/tmp).

Well, there's no reason to document it does, then.

Reply:
@Earnestly you are right, you can chroot to this path, [but when `makechrootpkg` drops root privileges](https://github.com/archlinux/devtools/blob/4c08847bfa7776abb884fef289a25c1cef9883fb/makechrootpkg.in#L216), suid bit is not preserved resulting in sudo error.
```
==> Installing missing dependencies...
sudo: effective uid is not 0, is /usr/bin/sudo on a file system with the 'nosuid' option set or an NFS file system without root privileges?
==> ERROR: 'pacman' failed to install missing dependencies.
```
[![asciicast](https://asciinema.org/a/iGap07zcOXkAkIOy7uVSmsGKY.svg)](https://asciinema.org/a/iGap07zcOXkAkIOy7uVSmsGKY)

Reply:
Ah so now the real problem shows itself which is attempting to use `sudo` in a `nosuid` mount.  In such a case can't you `-o remount,suid` beforehand since you have root priviledge?

Potentially dependencies can be brought into the chroot before entering it but I'm not sure of that workflow and how it interacts.

Reply:
> In such a case can't you `-o remount,suid` beforehand since you have root priviledge?
This is what is stated in my `man:arch-build -D` disclaimer we are talking about now.
```
       -D DIR, --directory=DIR
              The base directory for containers. Defaults to /var/lib/archbuild/<chroot-prefix>-<arch>.

              This directory usually contains a /root subdirectory that serves as template for user containers (named after $SUDO_USER, or /copy if unset).

              Mount point for DIR need to have nosuid option disabled.
              # findmnt -o TARGET,OPTIONS -T DIR
              To disable nosuid
              # mount -o remount,suid $(findmnt -no TARGET -T DIR)
```

Reply:
What I mean to suggest is can this be done outside of aurutils?  Ultimately aurutils is a collection of utilities that are intended to be used with custom scripts, ones such as which can set up chroots, change mount options, or whatever else.

Reply:
Yep, as mentioned by @AladW, you can create wrapper around `aur-utils` which e.g. creates/destroys costume `tmpfs` and pass it as argument for `-D` switch. But if we rely on user to do heavy lifting, we could at lest mention this in the manual, as this isn't an edge case but a common scenario to use `tmpfs` when building in chroot.

Reply:
Fair, knowing nothing else I'd agree that it might be worth mentioning in the `NOTES` manual.

Reply:
I can live with it being in `NOTES`, mentioning an edge case prominently in the option documentation directly is not something I'd do.